src/plugins/discover/public/application/main/state_management/discover_state.ts

```diff
@@ -372,6 +372,13 @@ export function getDiscoverStateContainer({
   };
 
   const transitionFromESQLToDataView = (dataViewId: string) => {
+    const dataView = internalStateContainer.getState().dataView;
+    if (dataView) {
+      // the most recent data view, built behind the scene when using ES|QL
+      // needs to be added to the list of adHocDataViews to be displayed correctly with the right icon
+      const prevAdHocDataViews = internalStateContainer.getState().adHocDataViews;
+      internalStateContainer.transitions.setAdHocDataViews([...prevAdHocDataViews, dataView]);
```

Comment:
Do you remember if we previously intentionally added the ES|QL data view as an ad hoc data view when switching back from ES|QL mode? I can't remember if it was an intentional decision or just a side effect of how we were handling things. Just thinking out loud if we _should_ be doing this or picking an available data view from the list instead.

Reply:
I don't remember, but from UX side it make sense, that I could just reuse the latest index pattern and ad-hoc data view? doesn't it? Also given it could be the case there are no data views available

Reply:
Yeah that's reasonable, especially in the no data view case. Ok, let's restore the same behaviour as before, at least for now 👍